File: src/HolidayManagement.Core/Models/HolidayRequest.cs
using System;

namespace HolidayManagement.Core.Models
{
    public class HolidayRequest
    {
        public int Id { get; set; }

        public DateTimeOffset StartDate { get; set; }

        public DateTimeOffset EndDate { get; set; }

        public HolidayRequestStatus Status { get; set; }

        public int EmployeeId { get; set; }
        
        public Employee Employee { get; set; }
    }
}

File: src/HolidayManagement.Client/ClientApp/src/app/models/holiday-request.ts
import { Employee } from "./employee";

export class HolidayRequest {
    id: number;
    startDate: string;
    endDate: string;
    status: HolidayRequestStatus;
    comments?: string;
    employeeId: number;
    employee?: Employee;
}

export enum HolidayRequestStatus {
    pending = 0,
    approved = 1,
    refused = 2
}